maa.jwt.verifier/src/jwt.cpp

```diff
@@ -0,0 +1,73 @@
+#include <jwt.hpp>
+
+#include <utils.hpp>
+#include <context.hpp>
+#include <base64.hpp>
+#include <regex>
+
+namespace mvj {
+
+    Jwt::Jwt() {}
+        
+    bool Jwt::deserialize(const std::string& token) {
+        try {
+            encoded_token_ = token;
+            std::vector<std::string> hps; 
+            strings::split(token, "\\.", hps);
+    
+            if (hps.size() != 3) {
+                throw std::runtime_error("Invalid token!");
+            }
+            encoded_header_ = hps[0];
+            encoded_payload_ = hps[1];
+            encoded_signature_ = hps[2];
+    
+            decoded_header_ = decode(encoded_header_);
+            decoded_payload_ = decode(encoded_payload_);
+    
+            jku_ = json::get_value(decoded_header_, "jku");
+            kid_ = json::get_value(decoded_header_, "kid");
+            attest_dns_ = parse_dns();
+            tenant_ = parse_tenant();
+        }
+        catch (const char* ex) {
```

Comment:
Did you mean to catch a char * here or an exception type?

Reply:
the intent is to catch everything. will fix it